test_bottles.py

```diff
@@ -1,90 +1,93 @@
 
 
-class BottlesTest(unittest.TestCase):
-    maxDiff = None
-
-    def test_the_first_verse(self):
-        expected = (
-            '99 bottles of beer on the wall, '
-            '99 bottles of beer.\n'
-            'Take one down and pass it around, '
-            '98 bottles of beer on the wall.\n'
-        )
-        self.assertEqual(Bottles().verse(99), expected)
-
-    def test_another_verse(self):
-        expected = (
-            '3 bottles of beer on the wall, '
-            '3 bottles of beer.\n'
-            'Take one down and pass it around, '
-            '2 bottles of beer on the wall.\n'
-        )
-        self.assertEqual(Bottles().verse(3), expected)
-
-    def test_verse_2(self):
-        expected = (
-            '2 bottles of beer on the wall, '
-            '2 bottles of beer.\n'
-            'Take one down and pass it around, '
-            '1 bottle of beer on the wall.\n'
-        )
-        self.assertEqual(Bottles().verse(2), expected)
-
-    def test_verse_1(self):
-        expected = (
-            '1 bottle of beer on the wall, '
-            '1 bottle of beer.\n'
-            'Take it down and pass it around, '
-            'no more bottles of beer on the wall.\n'
-        )
-        self.assertEqual(Bottles().verse(1), expected)
-
-    def test_verse_0(self):
-        expected = (
-            'No more bottles of beer on the wall, '
-            'no more bottles of beer.\n'
-            'Go to the store and buy some more, '
-            '99 bottles of beer on the wall.\n'
-        )
-        self.assertEqual(Bottles().verse(0), expected)
-
-    def test_a_couple_of_verses(self):
-        expected = (
-            '99 bottles of beer on the wall, '
-            '99 bottles of beer.\n'
-            'Take one down and pass it around, '
-            '98 bottles of beer on the wall.\n'
-            '\n'
-            '98 bottles of beer on the wall, '
-            '98 bottles of beer.\n'
-            'Take one down and pass it around, '
-            '97 bottles of beer on the wall.\n'
-        )
-        self.assertEqual(Bottles().verses(99, 98), expected)
-
-    def test_a_few_verses(self):
-        expected = (
-            '2 bottles of beer on the wall, '
-            '2 bottles of beer.\n'
-            'Take one down and pass it around, '
-            '1 bottle of beer on the wall.\n'
-            '\n'
-            '1 bottle of beer on the wall, '
-            '1 bottle of beer.\n'
-            'Take it down and pass it around, '
-            'no more bottles of beer on the wall.\n'
-            '\n'
-            'No more bottles of beer on the wall, '
-            'no more bottles of beer.\n'
-            'Go to the store and buy some more, '
-            '99 bottles of beer on the wall.\n'
-        )
-        self.assertEqual(Bottles().verses(2, 0), expected)
-
-    def test_the_whole_song(self):
-        expected = """99 bottles of beer on the wall, 99 bottles of beer.
+def test_the_first_verse():
```

Comment:
Later in the book we will be putting test suites for different classes in the same file. Is there a way of creating a separate namespace for the tests using Pytest?

Reply:
Yes - pytest enables grouping tests via classes and modules (files).
I believe you should be able to group them into the TestClasses to avoid namespace issues.

```
def test_method_a():
    setup_var = 5
    object = main.my_class()

    actual = object.act(setup_var)
    expected = 6

    assert actual == expected


def test_method_b():
    setup_var = 5
    object = main.my_class()

    actual = object.send(setup_var)
    expected = 6

    assert actual == expected
```

```
class TestMyClass:
    def test_method_a(self):
        setup_var = 5
        object = main.my_class()

        actual = object.act(setup_var)
       expected = 6

        assert actual == expected

    def test_method_b(self):
        setup_var = 5
        object = main.my_class()

        actual = object.send(setup_var)
        expected = 6

        assert actual == expected
```